test/unit/Validation/Constraint/ValidAtTestCase.php

```diff
@@ -0,0 +1,205 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation\Constraint;
+
+use DateInterval;
+use DateTimeImmutable;
+use Lcobucci\Clock\Clock;
+use Lcobucci\Clock\FrozenClock;
+use Lcobucci\JWT\Token\RegisteredClaims;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\ConstraintViolation;
+
+abstract class ValidAtTestCase extends ConstraintTestCase
```

Comment:
Common tests between `LooseValidAt` and `StrictValidAt` are now written once here.
Implementations are copy-pasted though, without a common base class.